protostar/commands/test/test_runner.py

```diff
@@ -122,11 +132,17 @@ async def _run_test_suite(
                         tx_info=call_result,
                     )
                 )
-            except ReportedException as err:
+            except ReportedException as ex:
                 self.queue.put(
                     FailedTestCase(
                         file_path=test_suite.test_path,
                         test_case_name=test_case_name,
-                        exception=err,
+                        exception=ex,
                     )
                 )
+
+    @staticmethod
+    def is_constructor_args_exception(ex: StarkException) -> bool:
+        if not ex.message:
+            return False
+        return "constructor" in ex.message and "__calldata_actual_size" in ex.message
```

Comment:
Won't this overlay with some `deploy_contract` exception?

Reply:
It won't because`deploy_contract` raises `StarknetRevertableException` and that exception keeps message in `details` property.  Added a test case for that.

Reply:
What if i try to deploy it using 0.9 deply syscall handler?

Reply:
Protostar throws `RevertableException` for deploy syscall error.